tests/ufunc_example/_module2.c

```diff
@@ -0,0 +1,69 @@
+#define NPY_NO_DEPRECATED_API NPY_1_7_API_VERSION
+
+#include <numpy/arrayobject.h>
+#include <numpy/ufuncobject.h>
+#include <Python.h>
+
+
+static double foo_inner(double a, double b)
+{
+    return a + b;
+}
+
+
+static void foo_loop(
+    char **args,
+    const npy_intp *dimensions,
+    const npy_intp *steps,
+    void *NPY_UNUSED(data)
+) {
+   const npy_intp n = dimensions[0];
+   for (npy_intp i = 0; i < n; i ++)
+   {
+       *(double *) &args[2][i * steps[2]] = foo_inner(
+       *(double *) &args[0][i * steps[0]],
+       *(double *) &args[1][i * steps[1]]);
+   }
+}
+
+
+static PyUFuncGenericFunction foo_loops[] = {foo_loop};
+static char foo_types[] = {NPY_DOUBLE, NPY_DOUBLE, NPY_DOUBLE};
+static void *foo_data[] = {NULL};
+static const char foo_name[] = "foo";
+static const char foo_docstring[] = ">>> foo(1, 2)\n3.0";
+
+static PyModuleDef moduledef = {
+    .m_base = PyModuleDef_HEAD_INIT,
+    .m_name = "_module2",
+    .m_size = -1
+};
+
+
+PyMODINIT_FUNC PyInit__module2(void); /* Silence -Wmissing-prototypes */
```

Comment:
Is this still necessary?

Reply:
It's just that I haven't had to do this double declaration (e.g., in pyerfa), and I cannot see any difference in the imports you make...

Reply:
It may not be necessary if you don't have warnings cranked all the way up. I'll try without.